app/views/public/babel_view.py

```diff
@@ -0,0 +1,34 @@
+from app.helpers.babel import babel
+from flask import request, Blueprint
+from config import LANGUAGES
+import json
+from flask import g
+
+app = Blueprint('babel', __name__)
+
+@app.route('/choose_language', methods=('POST',))
+def set_lang():
+    global selected_lang
+    l_code = request.form.get('l_code')
+    if l_code:
+        selected_lang = l_code
+    return selected_lang
+
+@app.route('/get_languages', methods=('GET',))
+def get_languages():
+	try:
+		selected_lang
+	except NameError:
+		output = {"selected" : LANGUAGES["en"], "all" : LANGUAGES}
+	else:
+		output = {"selected" : LANGUAGES[selected_lang], "all" : LANGUAGES}
+	return json.dumps(output)
+
+@babel.localeselector
+def get_locale():
+    try:
+        g.selected_lang = selected_lang
```

Comment:
Wait a second ... Now, if a user changes a language, the language changes for every other user using the application. You should store the selecting language in a store specific to a single user. My suggestion: use cookies to store the selected language.